### ISSUE TITLE
Low val mIoU

### ISSUE THREAD

**Issue by huqiming513** [yassouali/pytorch_segmentation#26]
I used the checkpoint of PSPNet downloaded from this repository, but the val mIoU as following is quite low at the beginning.
![image](https://user-images.githubusercontent.com/31566437/69404686-d4ef3e80-0d38-11ea-8139-5e7911a371e2.png)
Does anyone share the same problem with me?

**Comment by yassouali** [comment]
Hi, what dataset are you using it on ? does the mIoU stays this low after more that 10 iterations (I see you're using two epoch, so 20 images), because at the beginning, it is probable that some classes weren't seen, so the IoU for them is 0 and it reduces the mean.

**Comment by huqiming513** [comment]
Hi, glad to receive your reply! Yes, I'm using pascal voc dataset. This is the result after 1237 its, but I can't see it will show any increment... 
![image](https://user-images.githubusercontent.com/31566437/69509833-60edaa00-0f75-11ea-9cff-6b0124df02b6.png)

**Comment by huqiming513** [comment]
Oh, I changed the 'split' from 'train_aug' to 'trainval_aug' in config.json just now. The number of total samples is changed, but the mIoU is still at a low level. 
![image](https://user-images.githubusercontent.com/31566437/69511266-85985080-0f7a-11ea-9320-c4422bfd290f.png)

**Comment by yassouali** [comment]
After some  test, the problem is the low batch size, with only a batch of 2 the loss is a lot more noisy, and the batch norm's running params also change.

for example here are the results with a batch of 8:

![image](https://user-images.githubusercontent.com/45082182/69535761-e32d9c80-0f7c-11ea-8742-6ee8c9d2da4e.png)

Possible solutions, is to use a very low learning rate, set the batch norms training to false or of course a bigger batchsize.

**Comment by huqiming513** [comment]
> After some test, the problem is the low batch size, with only a batch of 2 the loss is a lot more noisy, and the batch norm's running params also change.
> 
> for example here are the results with a batch of 8:
> 
> ![image](https://user-images.githubusercontent.com/45082182/69535761-e32d9c80-0f7c-11ea-8742-6ee8c9d2da4e.png)
> 
> Possible solutions, is to use a very low learning rate, set the batch norms training to false or of course a bigger batchsize.

Thanks a lot. I've changed the batch size to 8 and run it on my lab's server. Now it works well. 
![image](https://user-images.githubusercontent.com/31566437/69538796-09bff780-0fbe-11ea-830d-7e497ef05c9b.png)
It occurs to me that the papers about norm methods I recently read, saying that the batch norm will show bad performance with small batch size especially with batch size of  exactly 2. 
![image](https://user-images.githubusercontent.com/31566437/69539402-5bb54d00-0fbf-11ea-94de-4a694eec1d39.png)
![image](https://user-images.githubusercontent.com/31566437/69539438-6ec81d00-0fbf-11ea-9f69-7c9befda80d0.png)

**Comment by yassouali** [comment]
Glad it worked, yes you're write, with low batch size, the batch norm running mean and var will change significantly.
